ui/packages/shared/profile/src/ProfileIcicleGraph/index.tsx

```diff
@@ -57,6 +64,46 @@ const ProfileIcicleGraph = ({
   const compareMode: boolean =
     selectQueryParam('compare_a') === 'true' && selectQueryParam('compare_b') === 'true';
   const {ref, dimensions} = useContainerDimensions();
+  const [storeSortBy, setStoreSortBy] = useURLState({param: 'sort_by', navigateTo});
+  const [localSortBy = FIELD_FUNCTION_NAME, setLocalSortBy] = useState(storeSortBy as string);
+  const setSortBy = useCallback(
+    (key: string) => {
+      setLocalSortBy(key);
+      setStoreSortBy(key);
+    },
+    [setLocalSortBy, setStoreSortBy]
+  );
+
+  const [storeGroupBy = [FIELD_FUNCTION_NAME], setStoreGroupBy] = useURLState({
+    param: 'group_by',
+    navigateTo,
+  });
+  const groupBy = useMemo(() => {
+    if (storeGroupBy !== undefined) {
+      if (typeof storeGroupBy === 'string') {
+        return [storeGroupBy];
+      }
+      return storeGroupBy as string[];
+    }
+    return [FIELD_FUNCTION_NAME];
+  }, [storeGroupBy]);
+
+  console.log('groupBy', storeGroupBy, groupBy);
+
+  const setGroupBy = useCallback(
+    (keys: string[]): void => {
+      setStoreGroupBy(keys);
+    },
+    [setStoreGroupBy]
+  );
+  const toggleGroupBy = useCallback(
+    (key: string): void => {
+      groupBy.includes(key)
+        ? setGroupBy(groupBy.filter(v => v !== key)) // remove
+        : setGroupBy([...groupBy, key]); // add
+    },
+    [groupBy]
```

Comment:
Is it intentional to not have the `setGroupBy` in the dependencies?

Reply:
Yes, and further down as well. If I add those functions/callbacks to the effect dependencies the component goes into an endless re-render. Happy to brainstorm how to fix it. Already tried some `useCallback` and `useMemo` approaches yesterday.

Reply:
Actually, adding `setGroupBy` seems to be fine. It's the `useEffect` further down that ends up looping.